### PR TITLE
Changed http requester

### PR DESCRIPTION
**Pull request by joserick**
Change of library Axios to native library Fetch based on problems with CORS #1 and defaults settings of Axios in Laravel

Advantage:
- Does not require dependencies
- Will not require future updates of axios
- It is completely native

### CODE REVIEW

**Comment by kinotto** [comment]
Hi @joserick,
can you please fix the unit tests under the /spec folder? 
thanks

**Comment by joserick** [comment]
Hi @kinotto 
I don't get any error from 'test-browser.html' but I have proceeded to remove axios, since it is no longer necessary.

Captura de pantalla de 2020-05-03 10-55-56

**Comment by joserick** [comment]
Hi @kinotto 

I just found out that 'Fetch' only works on web browser projects and not on node.js projects so I will try to implement the 'HTTPS request' that is a module native of node.js when this happens.

**Comment by joserick** [comment]
Hi @kinotto 
I have added 'HTTPS Request' as if it was Fetch to work natively on node.js projects; And in this way do not require any dependency.

Test:
Captura de pantalla de 2020-05-03 18-00-41

**Comment by kinotto** [comment]
hi @joserick, i see the changes but my tests are still failing

**Comment by kinotto** [comment]
Hi @joserick , i see the changes, but my tests are still failing, URLSearchParams is a constructor available only on the browser.
On top of that there could be some breaking changes in your PR, because axios and fetch have some slightly different behaviours.
As reported in the [doc](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#Differences_from_jQuery)

> The Promise returned from fetch() won’t reject on HTTP error status even if the response is an HTTP 404 or 500.

while this happens with axios 

image

**Comment by joserick** [comment]
Hi @kinotto 
Regarding URLSearchParams, node.js already has it in its [latest versions](https://nodejs.org/api/url.html#url_class_urlsearchparams) as a global object,  but I have added a couple of lines for previous versions of node.js

I have also managed to fix the HTTP 404/500 status error.

It is exactly what I am looking for, since in Laravel the Axios default configuration sends credentials and cookies without mentioning that Axios does not allow disabling CORS
> fetch() won't receive cross-site cookies. You can’t establish a cross site session using fetch(). Set-Cookie headers from other sites are silently ignored.
fetch() won’t send cookies, unless you set the credentials init option.

File: src/geonames.ts [hunk 1]
```diff
@@ -1,8 +1,36 @@
...
 import { baseParams, baseUri, baseUriCommercial, geoNamesAPI } from './geonames.config'
 
+if(typeof URLSearchParams === 'undefined'){
+  global.URLSearchParams = require('url').URLSearchParams;
+}
+
+if (typeof fetch !== 'function') {
+  const https = require('https')
+
+  global.fetch = async (url) => {
```
Comment (kinotto) [inline src/geonames.ts:12]: i would avoid polluting the global scope here to allow the user invoking their own version of fetch/UrlSearchParams
so i would rather put both functions declared locally inside the module.

**Pull request closed without merging**